ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/recovery/BasicXARecovery.java

```diff
@@ -202,7 +204,9 @@ private final JDBC2RecoveryConnection getConnection ()
 			if (dc != null)
 				dbProperties.put(TransactionalDriver.dynamicClass, dc);
 
-			return new JDBC2RecoveryConnection(theURL, dbProperties);
+			JDBC2RecoveryConnection connection = new JDBC2RecoveryConnection(theURL, dbProperties);
+			connections.add(connection);
```

Comment:
OK, so I will wait for you add another commit for JBTM-2911 before adding my approval.